src/utils/utilities.ts

```diff
@@ -240,3 +244,189 @@ export const calculateTimeLeft = (createdAt: string) => {
   // @ts-ignore
   return Math.max(0, Math.floor((targetTime - currentTime) / 1000));
 };
+
+export const generateDataFromPSBT = (base64Str: string, signer: Signer) => {
+  try {
+    const psbt = bitcoin.Psbt.fromBase64(base64Str);
+
+    // Convert the buffer to a hex string and reverse it to get the transaction ID
+    const txId = Buffer.from(psbt.txInputs[0].hash).reverse().toString('hex');
+    const vout = psbt.txInputs[0].index;
+
+    const changeAddress = getChangeAddress(base64Str);
+    const vBytes = estimateVByteFromPSBT(base64Str);
+
+    const signersList = [];
+    let signerMatched = false;
+
+    psbt.data.inputs.forEach((input) => {
+      if (input.bip32Derivation) {
+        // Loop through all derivations (in case there are multiple keys)
+        input.bip32Derivation.forEach((derivation) => {
+          const data = {
+            derivationPath: shortDerivationPath(derivation.path),
```

Comment:
You should probably not remove the path since what you have here is the pubkey itself, not xpub